opencensus/context/context.h

```diff
@@ -0,0 +1,69 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef OPENCENSUS_CONTEXT_CONTEXT_H_
+#define OPENCENSUS_CONTEXT_CONTEXT_H_
+
+#include <functional>
+#include <string>
+
+#include "opencensus/tags/tag_map.h"
+#include "opencensus/trace/span.h"
+
+namespace opencensus {
+namespace context {
+
+// Context holds information specific to an operation, such as a TagMap and
+// Span. Each thread has a currently active Context. Contexts are conceptually
+// immutable: the contents of a Context cannot be modified in-place.
+//
+// This is a draft implementation of Context, and we chose to depend on TagMap
+// and Span directly. In future, the implementation will change, so only rely
+// on the public API for manipulating Contexts. In future we may support
+// arbitrary keys and values.
+class Context {
+ public:
+  // Creates a default Context.
+  Context();
```

Comment:
Need this public? Please make it private until we need it.

Reply:
Done.